src/components/calcite-picker/calcite-picker.tsx

```diff
@@ -0,0 +1,291 @@
+
+  /**
+   * When true, the rows will be sortable via drag and drop.
+   * Only applies when mode is configuration
+   */
+  @Prop({ reflect: true }) dragEnabled = false;
+
+  @Prop({ reflect: true }) editEnabled = false; /* ignored unless mode is configuration */
+
+  /**
+   * Mode controls the presentation of the rows in their selected and deselected states.
+   * Selection mode shows either radio buttons or checkboxes depending on the value of multiple
+   * Configuration mode relies on a color highlight on the edge of the row for selected
+   * Mode must be set to configuration for drag and drop behavior to work.
+   */
+  @Prop({ reflect: true }) mode: "selection" | "configuration" = "selection";
+
+  /**
+   * Multpile Works similar to standard radio buttons and checkboxes.
+   * It also affects the presented icon when in Selection mode.
+   * When true, a user can select multiple rows at a time.
+   * When false, only a single row can be selected at a time,
+   * When false, selecting a new row will deselect any other selected rows.
+   */
+  @Prop({ reflect: true }) multiple = false;
+
+  /**
+   * The heading label for the entire Picker.
+   * Not to be confused with the heading for an individual row or for a sub-group of rows.
+   */
+  @Prop({ reflect: true }) textHeading: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @State() selectedValues: Set<object> = new Set();
+
+  @State() editing = false;
+
+  @Watch("editing")
+  editingChangeHandler() {
+    this.rows.forEach((item) => {
+      this.editing ? item.setAttribute("editing", "") : item.removeAttribute("editing");
+    });
+  }
+
+  deletedRows = new Set();
+
+  rows: any;
+
+  lastSelectedRow = null;
+
+  guid = `calcite-picker-${guid()}`;
+
+  observer = new MutationObserver(() => this.setupRows());
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.setupRows();
+  }
+
+  componentDidLoad() {
+    this.observer.observe(this.el, { childList: true, subtree: true });
+  }
+
+  componentDidUnload() {
+    this.observer.disconnect();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerSelectionChange: EventEmitter;
+
+  @Event() calcitePickerRowsDeleted: EventEmitter;
```

Comment:
should be named `calcitePickerItemDelete` if delete stuff is staying

Reply:
not staying.